File: SDKs/amazonaws.com/sagemaker/2017-07-24/go/pkg/models/operations/listprocessingjobs.go
package operations

import (
"openapi/pkg/models/shared")

type ListProcessingJobsQueryParams struct {
    MaxResults *string `queryParam:"style=form,explode=true,name=MaxResults"`
    NextToken *string `queryParam:"style=form,explode=true,name=NextToken"`
    
}


type ListProcessingJobsXAmzTargetEnum string

const (
    ListProcessingJobsXAmzTargetEnumSageMakerListProcessingJobs ListProcessingJobsXAmzTargetEnum = "SageMaker.ListProcessingJobs"
)


type ListProcessingJobsHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    XAmzTarget ListProcessingJobsXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
    
}

type ListProcessingJobsRequest struct {
    QueryParams ListProcessingJobsQueryParams 
    Headers ListProcessingJobsHeaders 
    Request shared.ListProcessingJobsRequest `request:"mediaType=application/json"`
    
}

type ListProcessingJobsResponse struct {
    ContentType string 
    ListProcessingJobsResponse *shared.ListProcessingJobsResponse 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/sagemaker/2017-07-24/typescript/src/sdk/models/operations/listprocessingjobs.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class ListProcessingJobsQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=MaxResults" })
  maxResults?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=NextToken" })
  nextToken?: string;
}

export enum ListProcessingJobsXAmzTargetEnum {
    SageMakerListProcessingJobs = "SageMaker.ListProcessingJobs"
}


export class ListProcessingJobsHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: ListProcessingJobsXAmzTargetEnum;
}


export class ListProcessingJobsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: ListProcessingJobsQueryParams;

  @SpeakeasyMetadata()
  headers: ListProcessingJobsHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.ListProcessingJobsRequest;
}


export class ListProcessingJobsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  listProcessingJobsResponse?: shared.ListProcessingJobsResponse;

  @SpeakeasyMetadata()
  statusCode: number;
}
